File: LC1318-MinimumFlipsToMakeAORBEqualToC.c
int minFlips(int a, int b, int c){
    uint mask = 1;
    int flips = 0;
    int bit_a, bit_b, bit_c;

    for (int i=0; i<sizeof(int) * 8; i++) {
        mask = (uint) 1 << i;
        bit_a = a & mask;
        bit_b = b & mask;
        bit_c = c & mask;
        if ((bit_a | bit_b) != bit_c) {
            if ((bit_a > 0) && (bit_b > 0)) {
                flips += 2;
            } else {
                flips += 1;
            }
        }
    }

    return flips;
}


// Simple C code solution with bitwise operations. Note we need to use uint for mask variable.

File: LC1318-MinimumFlipsToMakeAORBEqualToC.py
class Solution:
    def minFlips(self, a: int, b: int, c: int) -> int:
        flips = 0

        while a > 0 or b > 0 or c > 0:
            abit = a & 1
            bbit = b & 1
            cbit = c & 1

            # Check the bits and determine flips needed
            # abit bbit cbit flips
            #  0    0    0    0
            #  0    1    0    1
            #  1    0    0    1
            #  1    1    0    2
            #  0    0    1    1
            #  0    1    1    0
            #  1    0    1    0
            #  1    1    1    0
            if cbit == 0:
                flips += abit + bbit
            else:
                if abit == 0 and bbit == 0:
                    flips += 1

            a >>= 1
            b >>= 1
            c >>= 1

        return flips